node-api/backend/withdrawal.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package backend
+
+import (
+	"fmt"
+
+	"github.com/berachain/beacon-kit/errors"
+	"github.com/berachain/beacon-kit/node-api/handlers/beacon/types"
+	"github.com/berachain/beacon-kit/primitives/math"
+)
+
+func (b *Backend) PendingPartialWithdrawalsAtSlot(slot math.Slot) ([]*types.PendingPartialWithdrawalData, error) {
+	// Get the state at the given slot.
+	st, _, err := b.StateAtSlot(slot)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get state from slot %d", slot)
+	}
+
+	cTypePartialWithdrawals, err := st.GetPendingPartialWithdrawals()
```

Comment:
@nidhi-singh02: note that this can be nil pre-Electra. However I think this is safe as we only loop over it, which is safe to do with nil slices

Reply:
As we discussed, I see a comment on GetPendingPartialWithdrawals saying it will error if called before electra. The beacon API spec expected behaviour is to error out if the state is retrieved before pectra. 

If it errors out, then we will return early before the loop, so we should be good. 
We need to test this case out.

Reply:
Fixed